web/src/components/language-select/language-select.tsx

```diff
@@ -1,55 +1,73 @@
-import {
-  Localized,
-  withLocalization,
-  WithLocalizationProps,
-} from '@fluent/react';
-import Downshift from 'downshift';
-import * as React from 'react';
+export const ALL_LOCALES = 'all-locales';
 
-import './language-select.css';
-import useSortedLocales from '../../hooks/use-sorted-locales';
-import { useContributableLocales } from '../locale-helpers';
+import * as React from 'react';
+import {
+  LocalizedGetAttribute,
+  useAvailableLocales,
+  useLocale,
+} from '../locale-helpers';
+import { LabeledSelect } from '../ui/ui';
 
-export const ALL_LOCALES = 'all-locales';
+import '../localization-select/localization-select.css';
+import { Localized } from '@fluent/react';
 
-const LanguageSelect = ({
-  value,
-  onChange,
-  getString,
-}: {
+interface Props {
+  locale?: string;
+  onChange?: (props: string) => void;
   value: string;
-  onChange: (locale: string) => any;
-} & WithLocalizationProps) => {
-  const contributableLocales = useContributableLocales();
-  const [sortedLocales] = useSortedLocales(contributableLocales, getString);
+  includesAll?: boolean;
+}
+
+const LanguageSelect = ({ locale, onChange, value, includesAll }: Props) => {
+  const [clientLocale] = useLocale();
+  let languages: string[] = [];
+  const localesWithNames = useAvailableLocales();
+  const sortedLocales = localesWithNames.sort((a, b) =>
+    a.localeCompare(b, clientLocale)
+  );
+
+  languages = sortedLocales;
+
+  if (includesAll) {
+    languages = [ALL_LOCALES, ...languages];
+  }
+  // don't show select if we dont have multiple locales
+  if (localesWithNames.length <= 1) {
+    return null;
+  }
 
   return (
-    <Downshift initialInputValue={value} onChange={onChange}>
-      {({ getItemProps, highlightedIndex, isOpen, toggleMenu }) => (
-        <div className="language-select" onClick={() => toggleMenu()}>
-          <Localized id={value}>
-            <div className="current" />
-          </Localized>
-          {isOpen && (
-            <ul onClick={() => toggleMenu()}>
-              {[ALL_LOCALES].concat(sortedLocales).map((l, i) => (
-                <Localized key={l} id={l}>
-                  <li
-                    className={[
-                      l === value ? 'selected' : '',
-                      highlightedIndex === i ? 'highlighted' : '',
-                    ].join(' ')}
-                    {...getItemProps({ item: l })}
-                    value={l}
-                  />
-                </Localized>
-              ))}
-            </ul>
-          )}
-        </div>
+    <LocalizedGetAttribute id="localization-select" attribute="label">
+      {(label: string) => (
+        <LabeledSelect
+          style={{ height: 40, paddingTop: '5px', paddingBottom: '5px' }}
+          className="localization-select"
+          value={locale}
+          label={label}
+          isLabelVisuallyHidden={true}
```

Comment:
if a component is a boolean which is set to true, it's enough to just write the name of the prop without setting to true explicitly. For example,

```
<Component someBooleanPropWhichIsTrue />
```